File: src/robotpose.py
#!/usr/bin/env python

class RobotPose:

    def __init__(self, theta, x, y, leftWheel, rightWheel, time):
        self.theta = theta
        self.x = x
        self.y = y
        self.leftWheel = leftWheel
        self.rightWheel = rightWheel
        self.time = time

File: src/robotpose.cpp
#include <ros/ros.h>

class RobotPose {

    public:
        float theta;
        float x;
        float y;
        int leftWheel;
        int rightWheel;
        ros::Time time;

    RobotPose(float theta, float x, float y, int leftWheel, int rightWheel, ros::Time time) {
        this->theta = theta;
        this->x = x;
        this->y = y;
        this->leftWheel = leftWheel;
        this->rightWheel = rightWheel;
        this->time = time;
    }

    RobotPose(RobotPose* source, ros::Time time) {
        this->theta = source->theta;
        this->x = source->x;
        this->y = source->y;
        this->leftWheel = source->leftWheel;
        this->rightWheel = source->rightWheel;
        this->time = time;
    }
};
